lua/ide/components/explorer/component.lua

```diff
@@ -68,7 +69,14 @@ ExplorerComponent.new = function(name, config)
     local cwd = vim.fn.getcwd()
     local kind = vim.fn.getftype(cwd)
     local perms = vim.fn.getfperm(cwd)
-    local root = filenode.new(cwd, kind, perms, 0, self.config.list_directories_first)
+    local root = filenode.new(
```

Comment:
One request here, which I actually forgot to mention on the other PR. 

Can we follow this pattern here: https://github.com/ldelossa/nvim-ide/blob/572428fddc82281f57624d4541534621b93fc8e9/lua/ide/buffers/diffbuffer.lua#L79

can the new() method take an "opts" table which holds these two new options (and more as we develop) and merges the incoming opts with a default opts table? 

This way, we can avoid making these signatures larger and larger.

Reply:
Sure, done.